### PR TITLE
Minor interface updates

### PR DESCRIPTION
**Pull request by arch1995**
- adds `getUserInfo` method
- adds event emitters when connected or disconnected
- adds a `connected` flag.
- adds `state` object
- update deps
- update example + fix sign message response issue in web3.

### CODE REVIEW

File: src/Web3Auth.ts [hunk 1]
```diff
@@ -28,6 +41,8 @@ class Web3Auth implements IWeb3Auth {
```
Comment (chaitanyapotti) [inline src/Web3Auth.ts:41]: Pls also have a status variable

File: src/interface.ts [hunk 2]
```diff
@@ -85,8 +75,36 @@ export type AggregateVerifierParams = {
...
+  typeOfLogin: string;
+}
+
 export interface SessionData {
   privKey?: string;
+  userInfo?: SingleFactorUserInfo;
+}
+
+export interface IWeb3Auth {
+  sessionId: string | null;
```
Comment (chaitanyapotti) [inline src/interface.ts:98]: no need to expose this to the dapp right.. use status variable like in no-modal sdk

Comment (arch1995) [reply src/interface.ts:98]: we need it for SFA react-native

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**